### PR TITLE
verify ibm jvm compatability

### PR DESCRIPTION
**Pull request by ellisjoe**

### CODE REVIEW

**Comment by glentakahashi** [comment]
👍

File: build.gradle [hunk 1]
```diff
@@ -1,7 +1,7 @@
 buildscript {
     repositories {
-        jcenter()
         maven { url  "http://palantir.bintray.com/releases" }
+        jcenter()
```
Comment (uschi2000) [inline build.gradle:4]: critical change.

File: crypto-core/src/main/java/com/palantir/crypto2/cipher/Ciphers.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+     */
+    @VisibleForTesting
+    static String getProvider(List<String> providers) {
+        for (String provider : providers) {
+            if (Security.getProvider(provider) != null) {
+                return provider;
+            }
+        }
+        throw new IllegalStateException(
+                String.format("None of the requested JCE providers are acceptable: %s", providers));
```
Comment (uschi2000) [inline crypto-core/src/main/java/com/palantir/crypto2/cipher/Ciphers.java]: isn 't this: "None of the acceptable JCE providers are available" ?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,2 @@
+FROM ibmjava:8-sdk
```
Comment (uschi2000) [inline Dockerfile:1]: does this thing not contain SunJCE? (have you verified?)

File: crypto-core/src/main/java/com/palantir/crypto2/cipher/Ciphers.java (hunk 2 above)
Comment (ellisjoe) [reply crypto-core/src/main/java/com/palantir/crypto2/cipher/Ciphers.java]: done

File: build.gradle (hunk 1 above)
Comment (ellisjoe) [reply build.gradle:4]: sadly it is

File: Dockerfile (hunk 3 above)
Comment (ellisjoe) [reply Dockerfile:1]: removing "IBMJCE" from the list of acceptable providers gives me `java.lang.IllegalStateException: None of the acceptable JCE providers are acceptable: [SunJCE]` when running the tests in this container.

**Comment by uschi2000** [comment]
Ship it

Sent from Email+ secured by MobileIron

**Pull request merged**